src/Features/Core/Portable/RemoveUnnecessaryParentheses/AbstractRemoveUnnecessaryParenthesesDiagnosticAnalyzer.cs

```diff
@@ -116,9 +123,31 @@ private void AnalyzeSyntax(SyntaxNodeAnalysisContext context)
                 additionalLocations,
                 properties: null));
 
+            context.ReportDiagnostic(DiagnosticHelper.Create(
+                _unnecessaryWithoutSuggestionWithoutFadeDescriptor,
+                GetDiagnosticSquiggleLocation(parenthesizedExpression),
+                severity,
+                additionalLocations,
+                properties: null));
+
             context.ReportDiagnostic(Diagnostic.Create(
                 UnnecessaryWithoutSuggestionDescriptor,
                 parenthesizedExpression.GetLastToken().GetLocation(), additionalLocations));
         }
+
+        private Location GetDiagnosticSquiggleLocation(TParenthesizedExpressionSyntax parenthesizedExpression)
+        {
+            var parenthesizedExpressionLocation = parenthesizedExpression.GetLocation();
+            var expressionLineSpan = parenthesizedExpressionLocation.GetLineSpan();
+
+            if (expressionLineSpan.StartLinePosition.Line < expressionLineSpan.EndLinePosition.Line)
+            {
+                var lines = parenthesizedExpression.SyntaxTree.GetText().Lines;
+                var expressionFirstLine = lines.GetLineFromPosition(parenthesizedExpressionLocation.SourceSpan.Start);
+                var spanUntilEndOfLine = TextSpan.FromBounds(parenthesizedExpressionLocation.SourceSpan.Start, expressionFirstLine.Span.End);
+                return Location.Create(parenthesizedExpression.SyntaxTree, spanUntilEndOfLine);
+            }
+            return parenthesizedExpressionLocation;
```

Comment:
an easier way to express this is that you can just say that you want the span to go from the start of the parenthesizedExpr to the min of either it's endpoint, or the end of the line it's on.

Reply:
Thanks for the suggestion, that makes it much cleaner!